### DISCUSSION TITLE
Apple Universal Build

### DISCUSSION THREAD

**Discussion by chadwyk** [axmolengine/axmol#859] [General]
You have no idea how happy I am to find this project! I have always been a fan of Cocos2d since the days when it was cocos2d-iphone!  I had been really discouraged that the engine has been more or less abandoned by the Cocos team for the CocosCreator GUI. So this is great!

I see you changed it to support Apple Silicon (M1) without needing to find the libraries and link them myself (awesome!) I was able to get a project created for a new Apple Silicon game and it runs.  Is there a way to generate a Xcode project for a Universal build, or currently is the best method to generate two projects, then have to import my game code into both?  

Also is this the best forum to ask questions or is there another project site anywhere?

**Comment by paulocoutinhox** [comment]
Hi,

To support everything for iOS/tvOS/macOS in one Xcode project the best way is generate a xcframework.

The tutorial is here:
https://medium.com/swift-sections/xcframeworks-afd0b151d111

It need be done in prebuilt libraries repository:
https://github.com/axys1/buildware

After all libraries was built, need call those commands to generate the xcframework. I think that this is the first step.
